File: Medium/1701. Average Waiting Time/main.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        total_time = 0
        idle_time = 1
        for original_arrival, cooking_time in customers:
            total_time += cooking_time
            pending_time = idle_time - original_arrival
            if pending_time > 0:
                total_time += pending_time
                idle_time += cooking_time
            else:
                idle_time = original_arrival + cooking_time
        return total_time / len(customers)

File: Medium/1701. Average Waiting Time/main.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double total_time = 0;
        int idle_time = 1, pending_time, original_arrival, cooking_time;
        for (int i=0; i < customers.size(); i++) {
            original_arrival = customers[i][0];
            cooking_time = customers[i][1];
            total_time += cooking_time;
            pending_time = idle_time - original_arrival;
            if (pending_time > 0) {
                total_time += pending_time;
                idle_time += cooking_time;
            }
            else {
                idle_time = original_arrival + cooking_time;
            }
        }
        return total_time / customers.size();
    }
};
